src/Microsoft.Azure.SignalR.Protocols/CheckWithAckMessage.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.SignalR.Protocol
+{
+    /// <summary>
+    /// Base class of check-with-ack messages between Azure SignalR Service and SDK.
+    /// </summary>
+    public abstract class CheckWithAckMessage : ExtensibleServiceMessage, IAckableMessage, IMessageWithTracingId
+    {
+        /// <summary>
+        /// Gets or sets the ack id.
+        /// </summary>
+        public int AckId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the tracing Id
+        /// </summary>
+        public ulong? TracingId { get; set; }
+
+        protected CheckWithAckMessage(int ackId, ulong? tracingId)
+        {
+            AckId = ackId;
+            TracingId = tracingId;
+        }
+    }
+
+    /// <summary>
+    /// A waiting for ack check-user-in-group message.
```

Comment:
~~waiting for~~?

Reply:
https://github.com/Azure/azure-signalr/blob/dev/src/Microsoft.Azure.SignalR.Protocols/GroupMessage.cs#L252
old comment

I think it should be `waiting-for-ack`